client/src/pages/ContestDetails.js

```diff
@@ -189,6 +226,11 @@ export default function ContestDetails(props) {
         // eslint-disable-next-line
     }, [])
 
+    useEffect(() => {
+        //when payment dialog close, should fetch new data 
+        getContestInfo(contestId)
```

Comment:
are you getting a warning that `contestId` needs to be part of this dependency ?

Reply:
Yep, there was a warning, I just add contestId into the dependency array